File: C++/arrange.cpp
void Solution::arrange(vector<int> &Vec) {
    // Do not write main() function.
    // Do not read input, instead use the arguments to the function.
    // Do not print the output, instead return values as specified
    // Still have a doubt. Checkout www.interviewbit.com/pages/sample_codes/ for more details
    int N = Vec.size();
    for(int i = 0; i < N; ++i) {
        Vec[i] = Vec[i] + (Vec[Vec[i]]%N) * N;
    }

    for(int i = 0; i < N; ++i) {
        Vec[i] = Vec[i] / N;
    }
}

File: Python/arrange.py
class Solution:
    # @param A : list of integers
    def arrange(self, A):
        n = len(A)
        B = [0]*n
        for i in range(n):
            B[i] = A[i]
        for i in range(n):
            A[i] = B[B[i]]
        
